### PR TITLE
Feature: Support ConstantFP.getInfinity

### PR DESCRIPTION
**Pull request by ovr**
Hey!

Adds the `ConstantFP` method [getInfity](http://llvm.org/doxygen/classllvm_1_1ConstantFP.html#a6f2864671b22c9f6135b6ae5d41bcf54).

Thanks

### CODE REVIEW

File: src/ir/constant-fp.cc [hunk 1]
```diff
@@ -51,6 +51,17 @@ NAN_METHOD(ConstantFPWrapper::getNaN) {
     info.GetReturnValue().Set(ConstantWrapper::of(nan));
 }
 
+NAN_METHOD(ConstantFPWrapper::getInfinity) {
+    if (info.Length() != 1 || !TypeWrapper::isInstance(info[0])) {
```
Comment (MichaReiser) [inline src/ir/constant-fp.cc]: Could you please add support for the optional negative case as well [getInfity](http://llvm.org/doxygen/classllvm_1_1ConstantFP.html#a6f2864671b22c9f6135b6ae5d41bcf54)?

Comment (ovr) [reply src/ir/constant-fp.cc]: Yes, sure!

File: llvm-node.d.ts [hunk 2]
```diff
@@ -178,7 +178,7 @@ declare namespace llvm {
 
     static getNaN(type: Type): Constant;
 
-    static getInfinity(type: Type): Constant;
+    static getInfinity(type: Type, negative?: boolean/* = false */): Constant;
```
Comment (ovr) [inline llvm-node.d.ts]: Because:

image

**Comment by ovr** [comment]
@MichaReiser can you take a look again? Thanks

File: src/ir/constant-fp.cc [hunk 3]
```diff
@@ -52,12 +52,22 @@ NAN_METHOD(ConstantFPWrapper::getNaN) {
...
+        (info.Length() > 2 || info.Length() == 0)
+        || !TypeWrapper::isInstance(info[0])
+        || (info.Length() == 2 && !info[1]->IsBoolean())
+    ) {
+        return Nan::ThrowTypeError("getInfinity needs to be called with: type: Type, negative?: boolean = false");
+    }
+
+    bool negative = false;
+
+    if (info.Length() == 3) {
```
Comment (MichaReiser) [inline src/ir/constant-fp.cc]: Shouldn't this be `info.Length() == 2` or just `info.Length() > 1`?

Comment (MichaReiser) [reply src/ir/constant-fp.cc]: No worry. Happens to anyone and isn't that what code reviews are for :)

**Pull request merged**